### PR TITLE
Support deprecated v1 clara version for backward comptability

### PR DESCRIPTION
**Pull request by SachidanandAlle**
For more details:
https://github.com/NVIDIA/ai-assisted-annotation-client/issues/59#issuecomment-592296387

### CODE REVIEW

**Comment by lassoan** [comment]
Thanks for the quick update. I'll test this now.

**Comment by lassoan** [comment]
It works very well.

One important issue to fix: Default server name address should not be displayed in the server address combo box and should not be saved into application settings. The module should just replace empty text by the default server address, but not write it back to settings or GUI widget.

You can add this line to setupOptionsFrame method to make this clear to users:

        self.ui.serverComboBox.lineEdit().setPlaceholderText ("enter server address or leave empty to use default")

Another minor thing. Since segmentation tools are independent (they are not meant to be used together, such as starting segmentation with one tool then refine it with another), it would be better to allow opening only one section at a time, by adding something like this to setupOptionsFrame:

```
self.collapsibleButtonsGroup = qt.QButtonGroup()
        self.collapsibleButtonsGroup.addButton(self.ui.segmentationCollapsibleButton)
        self.collapsibleButtonsGroup.addButton(self.ui.annotationCollapsibleButton)
        self.collapsibleButtonsGroup.addButton(self.ui.deepgrowCollapsibleButton)
        self.ui.segmentationCollapsibleButton.collapsed = False  # show the first one by default; instead we could show the first one that has available models 
```

I would like to update our server to v2 interface and upload DeepGrow models. Can you point me to documentation that describes the install process? I tried it a few months ago and I succeeded with installation of v1 server and models but I could not make v2 server work (or could not upload models, I don't remember exactly). Thank you!

**Comment by SachidanandAlle** [comment]
For default server, I guess we have to show it to end user as part of upload agreement?  Not true?

https://github.com/NVIDIA/ai-assisted-annotation-client/blob/cd52c00934178a3e5371240ab5ff2b6f4a6496af/slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py#L263-L268


For second point,  segmentation & dextr3d are kind of together.. from segmentation, you get the extreme points.. and which will be your initial inputs to dextr3d..  i mean segmentation forwards it's feed to dextra3d

however deepgrow can be independent of segmentation and dextra3d.. i can fix that way..

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 1]
```diff
@@ -149,17 +147,21 @@ def currentSegmentID(self):
...
         self.logic.setServer(self.serverUrl())
-        self.logic.setUseCompression(slicer.util.settingsValue("NVIDIA-AIAA/compressData",
-                                                               True,
-                                                               converter=slicer.util.toBool))
+        self.logic.setUseCompression(slicer.util.settingsValue(
+            "NVIDIA-AIAA/compressData",
+            True, converter=slicer.util.toBool))
+        self.logic.setUseSession(slicer.util.settingsValue(
+            "NVIDIA-AIAA/aiaaSession",
+            True, converter=slicer.util.toBool))
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: do we want default to be false here also?

**Comment by lassoan** [comment]
Yes, we would still show the server address in the warning popup.

I did not know about potential cooperation between segmentation and dextr3d. I guess I'll have to try it. If the modes are not completely independent then it is probably better to leave them as is (no forcing of showing only one is needed).

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 2]
```diff
@@ -389,21 +390,24 @@ def onClickSegmentation(self):
...
             slicer.util.errorDisplay(operationDescription + " - unexpected error.", detailedText=traceback.format_exc())
-            return
         finally:
             qt.QApplication.restoreOverrideCursor()
             self.progressBar.hide()
 
         msg = 'Run segmentation for ({0}): {1}\t\n' \
               'Time Consumed: {2:3.1f} (sec)'.format(model, result, (time.time() - start))
         logging.info(msg)
-        self.onClickEditAnnotationFiducialPoints()
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: why do we remove this,
or we put it somewhere else?

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 3]
```diff
@@ -660,7 +670,7 @@ def setMRMLDefaults(self):
         self.scriptedEffect.setParameterDefault("ModelFilterLabel", "")
         self.scriptedEffect.setParameterDefault("SegmentationModel", "")
         self.scriptedEffect.setParameterDefault("AnnotationModel", "")
-        self.scriptedEffect.setParameterDefault("AnnotationModelFiltered", 1)
+        self.scriptedEffect.setParameterDefault("AnnotationModelFiltered", 0)
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: what does this change combine with line 792 gives us

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py (hunk 2 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: fixed

Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: it's moved after success

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py (hunk 1 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: fixed

**Comment by SachidanandAlle** [comment]
Added snapshot + steps for deepgrow in the readme..
@lassoan if all ok.. please approve..

hopefully we can publish the plugin with these changes in next nightly build

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py (hunk 3 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: by default we don't want enable it.. couple of review comments from https://github.com/NVIDIA/ai-assisted-annotation-client/issues/59#issuecomment-592296387

**Pull request merged**